node/node.go

```diff
@@ -371,6 +371,25 @@ func (n *Node) GetESDTData(address, tokenID string, nonce uint64) (*esdt.ESDigit
 		return nil, err
 	}
 
+	if core.ESDTType(esdtToken.Type) != core.Fungible {
+		if esdtToken.TokenMetaData == nil {
+			// TODO: analyze why n.getSystemAccount() doesn't work here. The difference is that here we use AccountsAdapter
```

Comment:
because the n.stateComponents.AccountsAdapter() will always hold a valid and loadable trie pointer due to the normal block processing.  n.stateComponents.AccountsAdapterAPI is not "maintained" automatically, so you will need to call a RecreateTrie operation. Strongly suggest to merge the branch `api-calls-fixes-for-scheduled-roothash` in this one and call an extra instruction on the n.getSystemAccount() function. Read below

Reply:
I do not like this resolve - we should use blockchainhook like resolve - all import the interface from vm-common. which resolves this code atuomatically.

Reply:
used esdt storage handler